Rocket.Chat/Extensions/UIBarButtonItem+Badge.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  UIBarButtonItem+Badge.swift
+//  Rocket.Chat
+//
+//  Created by Lucas Woo on 4/26/17.
+//  Copyright © 2017 Rocket.Chat. All rights reserved.
+//
+
+import Foundation
+import ObjectiveC
```

Comment:
I don't think a custom view can resolve this issue much more clearly since the left navigation button and it's target actions are created in an UINavigationController extension in SideMenuController library. And the badge extension is loaded lazily, so I think it will not impact existing components' performance.